sdl-core/src/main/scala/io/smartdatalake/workflow/action/spark/customlogic/CustomDsNto1Transformer.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2022 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.smartdatalake.workflow.action.spark.customlogic
+
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.util.spark.EncoderUtil
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+
+import scala.reflect.runtime.universe.typeOf
+
+/**
+ * Interface to define a custom Spark-Dataset transformation (1:1)
+ * When you implement this interface, you need to provide two case classes: One for your input Dataset
+ * and one for your output Dataset.
+ */
+trait CustomDsNto1Transformer extends Serializable {
+
+  private[smartdatalake] def transformBasedOnDataObjectId(session: SparkSession, options: Map[String, String], dfs: Map[String, DataFrame]): DataFrame = {
+
+    // lookup transform method
+    val methodName = "transform"
+    val mirror = scala.reflect.runtime.currentMirror
+    val typeTagSubclass = mirror.classSymbol(this.getClass).toType
+    val transformMethodsOfSubclass = typeTagSubclass.members.filter(_.isMethod).filter(_.name.toString == methodName)
```

Comment:
What I find is incredible is that the user's transform method is never directly called now, we are finding it with reflection :D We are doing assembler-style programming with scala :P

Reply:
Yes, we already use similar "patterns" in ConfigParser...